apis/duck/v1/kresource_type.go

```diff
@@ -33,6 +33,8 @@ type KRShaped interface {
 	GetTypeMeta() *metav1.TypeMeta
 
 	GetStatus() *Status
+
+	GetHappyConditionType() apis.ConditionType
```

Comment:
I would call this GetTopLevelConditionType

happy was a tongue  and cheek joke for the internal code.